src/components/Market/MarketWidgets.tsx

```diff
@@ -0,0 +1,33 @@
+import classNames from "classnames";
+import React from "react";
+
+export type Widget = {
+    component: React.ComponentType; 
+    label: string | JSX.Element;
+};
+
+type MarketWidgetsProps = {
+    title: string;
+    widgetsList: Widget[];
+    className: string
+};
+
+const MarketWidgets = ({ title, widgetsList, className}: MarketWidgetsProps): JSX.Element => {
+    const titleClassnames = classNames("market__title-little ", className);
+    return(
+        <>
+            <h3 className={titleClassnames}>{title}</h3>
+            <div className="market__images">
+                {widgetsList.map(({ component: Component, label }, index) => (
```

Comment:
```
import React, { ReactNode } from "react";
import classes from './MarketWidgets.module.scss';

export type WidgetItem = {
    id: string;
    component: ReactNode; 
    label: ReactNode;
};

type MarketWidgetsProps = {
    title: string;
    widgetsList: WidgetItem[];
};

const MarketWidgets = ({ title, widgetsList }: MarketWidgetsProps): JSX.Element => (
    //если не переименовывать и возвращать {component} ошибка: 
    //Functions are not valid as a React child. 
    //This may happen if you return a Component instead of <Component /> from render. 
    <>
        <h3 className={classes["market__title-little"]}>{title}</h3>
        <div className={classes["market__images"]}>
            {widgetsList.map(({ id, component: Component, label }) => (
                <div className={classes["market__images-block"]} key={id}>
                    <Component /> 
                    <span className={classes["market__images-span"]}>{label}</span>
                </div>
            ))}
        </div>
    </>
);

export default MarketWidgets;
```
аналогичная ошибка тут

Reply:
Если разберешься с первым случаем, то здесь будет легко исправить. Здесь чуть более запутанно, чтоб разобраться.